packages/common-sdk/src/web3/token-util.ts

```diff
@@ -42,4 +45,59 @@ export class TokenUtil {
 
     return accountInfo;
   };
+
+  /**
+   * Create an ix to send a spl-token to another wallet.
+   * This function will handle the associated token accounts internally.
+   * SOL is wrapped, sent as a spl-token and cleaned up after the transfer.
+   *
+   * @param connection - Connection object
+   * @param sourceWallet - PublicKey for the sender's wallet
+   * @param destinationWallet - PublicKey for the receiver's wallet
+   * @param tokenMint - Mint for the token that is being sent.
+   * @param tokenDecimals - Decimal for the token that is being sent.
+   * @param amount - Amount of token to send
+   * @param getAccountRentExempt - Fn to fetch the account rent exempt value
+   * @param payer - PublicKey for the payer that would fund the possibly new token-accounts. (must sign the txn)
+   * @returns
+   */
+  static async createSendTokensToWalletInstruction(
+    connection: Connection,
+    sourceWallet: PublicKey,
+    destinationWallet: PublicKey,
+    tokenMint: PublicKey,
+    tokenDecimals: number,
+    amount: u64,
+    getAccountRentExempt: () => Promise<number>,
+    payer?: PublicKey
+  ): Promise<Instruction> {
+    invariant(!amount.eq(ZERO), "SendToken transaction must send more than 0 tokens.");
+
+    const sourceTokenAccount = await deriveATA(sourceWallet, tokenMint);
```

Comment:
When sending SOL, doesn't the source side also need to use resolveOrCreateATA?
If so, since swapping SOL withDevFee will create/delete 3 WSOL accounts and increase signatures, why not just send SOL directly?